framework/doc/content/source/meshgenerators/XYDelaunayGenerator.md

```diff
@@ -55,6 +55,8 @@ simple Laplacian smoother; this can have unwanted side-effects on
 meshes with concave boundaries or poor nodal valences, and so it is
 disabled by default for robustness.
 
+Interior vertices can be specified using [!param](/Mesh/XYDelaunayGenerator/interior_points). This is useful when interior nodal locations need to be specified. If [!param](/Mesh/XYDelaunayGenerator/smooth_triangulation) is enabled, these points will be shifted.
```

Comment:
Do we have any test coverage for the previous mechanism for specifying interior nodes: via the input mesh?  That was apparently so non-intuitive that I only know of users who employed it *by accident*, so we might want to get a regression test for it and document it here.  Adding the documentation just for `interior_points` suggests (in an "exception that proves the rule" sense) that this new option (which I do like; much friendlier) is the *only* way to specify interior nodes.

We should also make it clear, not just that `smooth_triangulation` will still move those points, but that enabling mesh refinement will still add more.

Reply:
> Do we have any test coverage for the previous mechanism for specifying interior nodes: via the input mesh?

I think this is what you are describing: `test/tests/meshgenerators/xy_delaunay_generator/xydelaunay_from_2d.i`
I didn't know it had that functionality, which is why the docs sugest it's the only way! I thought the input mesh only defined the boundary and not the interior.

Also, could I use that functionality for an exodus mesh that I loaded in that has hanging nodes. I know we don't preserve the information that would allow us to run a simulation on the mesh with those hanging nodes but what if I triangulate that mesh. When I try I get the error below. Before I do some digging wasn't sure if there was an immediate pitfall to prevent triangulating a hanging node mesh.

```
*** ERROR ***
Bad edge topology found by MeshedHole
Stack frames: 18
0: libMesh::print_trace(std::ostream&)
1: libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*, std::ostream&)
2: /data/nezdmn/miniforge/envs/moose/libmesh/lib/libmesh_opt.so.0(+0xca0a6a) [0x7f919b7f6a6a]
3: libMesh::TriangulatorInterface::MeshedHole::MeshedHole(libMesh::MeshBase const&, std::set<unsigned long, std::less<unsigned long>, std::allocator<unsigned long> >)
4: libMesh::TriangulatorInterface::elems_to_segments()
5: libMesh::Poly2TriTriangulator::triangulate()
6: XYDelaunayGenerator::generate()
7: MeshGenerator::generateInternal()
8: MeshGeneratorSystem::executeMeshGenerators()
9: Action::timedAct()
10: ActionWarehouse::executeActionsWithAction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
11: ActionWarehouse::executeAllActions()
12: MooseApp::runInputFile()
13: MooseApp::run()
```

Reply:
Let's create a parameter group for these options for specifying interior points

Reply:
I brought this up closer to the top of the documentation where we talk about the input mesh defining boundary and interior points.